packages/gensx-storage/src/blob/types.ts

```diff
@@ -0,0 +1,241 @@
+
+  /**
+   * Get a readable stream of the blob's content.
+   * @returns A readable stream of the blob's content.
+   */
+  getStream(): Promise<Readable>;
+
+  /**
+   * Put JSON data into the blob.
+   * @param value The JSON data to store.
+   * @param options Optional metadata and etag for conditional updates.
+   * @returns The etag of the stored blob.
+   */
+  putJSON(value: T, options?: BlobOptions): Promise<{ etag: string }>;
+
+  /**
+   * Put string data into the blob.
+   * @param value The string data to store.
+   * @param options Optional metadata and etag for conditional updates.
+   * @returns The etag of the stored blob.
+   */
+  putString(value: string, options?: BlobOptions): Promise<{ etag: string }>;
+
+  /**
+   * Put raw data into the blob with metadata.
+   * @param value The data to store.
+   * @param options Optional metadata and etag for conditional updates.
+   * @returns The etag of the stored blob.
+   */
+  putRaw(value: Buffer, options?: BlobOptions): Promise<{ etag: string }>;
+
+  /**
+   * Put a readable stream into the blob.
+   * @param stream The readable stream to store.
+   * @param options Optional metadata and etag for conditional updates.
+   * @returns The etag of the stored blob.
+   */
+  putStream(stream: Readable, options?: BlobOptions): Promise<{ etag: string }>;
+
+  /**
+   * Delete the blob.
+   */
+  delete(): Promise<void>;
+
+  /**
+   * Check if the blob exists.
+   * @returns True if the blob exists, false otherwise.
+   */
+  exists(): Promise<boolean>;
+
+  /**
+   * Get metadata associated with the blob
+   */
+  getMetadata(): Promise<Record<string, string> | null>;
+
+  /**
+   * Update metadata associated with the blob
+   * @param metadata The new metadata to store
+   */
+  updateMetadata(metadata: Record<string, string>): Promise<void>;
+}
+
+/**
+ * Interface for blob storage
+ */
+export interface BlobStorage {
+  /**
+   * Get a blob object for a specific key
+   */
+  getBlob<T>(key: string): Blob<T>;
+
+  /**
+   * List all blobs with the given prefix
+   */
+  listBlobs(prefix?: string): Promise<string[]>;
+}
+
+/**
+ * Provider configuration kinds
+ */
+export type StorageKind = "filesystem" | "cloud";
+
+/**
+ * Base provider props
+ */
+export interface BaseBlobProviderProps {
+  /**
+   * Storage kind
+   */
+  kind: StorageKind;
+
+  /**
+   * Default prefix for all blob keys
+   */
+  defaultPrefix?: string;
+}
+
+/**
+ * Filesystem provider props
+ */
+export interface FileSystemBlobProviderProps extends BaseBlobProviderProps {
+  kind: "filesystem";
+
+  /**
+   * Root directory for storing blobs
+   */
+  rootDir?: string;
+}
+
+/**
+ * Cloud provider props
+ */
+export interface CloudBlobProviderProps extends BaseBlobProviderProps {
+  kind: "cloud";
+
+  /**
+   * Optional organization ID override (default: uses current org from context)
+   */
+  organizationId?: string;
+}
+
+/**
+ * Union type for blob provider props
+ */
+export type BlobProviderProps =
```

Comment:
I do think we should have `kind` be optional, and do the right thing when running locally/somewhere vs our cloud..

I guess its hard because we need the user to tell us where to put files for local execution. Can we overcome that by defaulting to the `.gensx` directory in the project?

Reply:
i like this idea. i'll go add it in. Seems like that would fix the issue with having to specify "kind"

Reply:
i guess the only tricky thing is we would need a way to identify if it's being run in our cloud or not.

Reply:
We can do that with an environment variable.

Reply:
We should do this before we onboard anyone. I am fine with merging without, but let's be sure to do it before we move onto other workstreams.